tests/CallbackScheduler_test.cdc

```diff
@@ -0,0 +1,402 @@
+        ),
+
+        // Valid cases - should return EstimatedCallback with no error
+        EstimateTestCase(
+            name: "High priority effort",
+            timestamp: futureTime + 1.0,
+            priority: CallbackScheduler.Priority.High,
+            executionEffort: 5000,
+            data: nil,
+            expectNil: false,
+            expectedFee: 5.0,
+            expectedTimestamp: futureTime + 1.0,
+            expectedError: nil
+        ),
+        EstimateTestCase(
+            name: "Medium priority minimum effort",
+            timestamp: futureTime + 4.0,
+            priority: CallbackScheduler.Priority.Medium,
+            executionEffort: 5,
+            data: nil,
+            expectNil: false,
+            expectedFee: 0.0025,
+            expectedTimestamp: futureTime + 4.0,
+            expectedError: nil
+        ),
+        EstimateTestCase(
+            name: "Far future timestamp",
+            timestamp: farFutureTime,
+            priority: CallbackScheduler.Priority.High,
+            executionEffort: 1000,
+            data: nil,
+            expectNil: false,
+            expectedFee: 1.0,
+            expectedTimestamp: farFutureTime,
+            expectedError: nil
+        ),
+        EstimateTestCase(
+            name: "String data",
+            timestamp: futureTime + 10.0,
+            priority: CallbackScheduler.Priority.High,
+            executionEffort: 1000,
+            data: "string data",
+            expectNil: false,
+            expectedFee: 1.0,
+            expectedTimestamp: futureTime + 10.0,
+            expectedError: nil
+        ),
+        EstimateTestCase(
+            name: "Dictionary data",
+            timestamp: futureTime + 11.0,
+            priority: CallbackScheduler.Priority.Medium,
+            executionEffort: 1000,
+            data: {"key": "value"},
+            expectNil: false,
+            expectedFee: 0.5,
+            expectedTimestamp: futureTime + 11.0,
+            expectedError: nil
+        )
+    ]
+
+    for testCase in estimateTestCases {
+        runEstimateTestCase(testCase: testCase)
+    }
+}
+
+access(all) fun runEstimateTestCase(testCase: EstimateTestCase) {
+    let result = CallbackScheduler.estimate(
+        data: testCase.data,
+        timestamp: testCase.timestamp,
+        priority: testCase.priority,
+        executionEffort: testCase.executionEffort
+    )
+    
+    if testCase.expectNil {
+        Test.assert(result == nil, message: "expected nil for test case: ".concat(testCase.name))
+    } else {
+        Test.assert(result != nil, message: "expected non-nil for test case: ".concat(testCase.name))
+        
+        if let estimate = result {
+            // Check fee
+            if let expectedFee = testCase.expectedFee {
+                Test.assert(expectedFee == estimate.flowFee, message: "fee mismatch for test case: ".concat(testCase.name))
```

Comment:
Not sure if the test framework already got updated to the version of Cadence that introduced the feature: Instead of using `.concat`, you can use string templates (here and elsewhere):
```suggestion
                Test.assert(expectedFee == estimate.flowFee, message: "fee mismatch for test case: \(testCase.name)")
```